resources/windows/wix/customactions.wxs.erb

```diff
@@ -217,5 +217,99 @@ End If
+        Execute="immediate"/>
+    <CustomAction
+        Id="GrantPuppetPermissions"
+        BinaryKey="WixCA"
+        DllEntry="<%= @platform.architecture == "x64" ? "WixQuietExec64" : "WixQuietExec" %>"
+        Execute="deferred"
+        Return="check"
+        Impersonate="no"/>
+
+    <!-- Leave root APPDATADIR as-is -->
+    <CustomAction
+        Id="SetResetCodePermissions"
+        Property="ResetCodePermissions"
+        Value="&quot;[%WINDIR]\System32\cmd.exe&quot; /c IF EXIST &quot;[%ProgramData]\PuppetLabs\code&quot; &quot;[%WINDIR]\System32\icacls.exe&quot; &quot;[%ProgramData]\PuppetLabs\code\*&quot; /reset /T /C"
+        Execute="immediate"/>
+    <CustomAction
+        Id="ResetCodePermissions"
+        BinaryKey="WixCA"
+        DllEntry="<%= @platform.architecture == "x64" ? "WixQuietExec64" : "WixQuietExec" %>"
+        Execute="deferred"
+        Return="check"
+        Impersonate="no"/>
+
+    <CustomAction
+        Id="SetResetFacterPermissions"
+        Property="ResetFacterPermissions"
+        Value="&quot;[%WINDIR]\System32\cmd.exe&quot; /c IF EXIST &quot;[%ProgramData]\PuppetLabs\facter&quot; &quot;[%WINDIR]\System32\icacls.exe&quot; &quot;[%ProgramData]\PuppetLabs\facter\*&quot; /reset /T /C"
+        Execute="immediate"/>
+    <CustomAction
+        Id="ResetFacterPermissions"
+        BinaryKey="WixCA"
+        DllEntry="<%= @platform.architecture == "x64" ? "WixQuietExec64" : "WixQuietExec" %>"
+        Execute="deferred"
+        Return="check"
+        Impersonate="no"/>
+
+    <CustomAction
+        Id="SetResetPxpAgentPermissions"
+        Property="ResetPxpAgentPermissions"
+        Value="&quot;[%WINDIR]\System32\cmd.exe&quot; /c IF EXIST &quot;[%ProgramData]\PuppetLabs\pxp-agent&quot; &quot;[%WINDIR]\System32\icacls.exe&quot; &quot;[%ProgramData]\PuppetLabs\pxp-agent\*&quot; /reset /T /C"
+        Execute="immediate"/>
+    <CustomAction
+        Id="ResetPxpAgentPermissions"
+        BinaryKey="WixCA"
+        DllEntry="<%= @platform.architecture == "x64" ? "WixQuietExec64" : "WixQuietExec" %>"
+        Execute="deferred"
+        Return="check"
+        Impersonate="no"/>
+
+    <CustomAction
+        Id="SetResetPuppetPermissions"
+        Property="ResetPuppetPermissions"
+        Value="&quot;[%WINDIR]\System32\cmd.exe&quot; /c IF EXIST &quot;[%ProgramData]\PuppetLabs\puppet&quot; &quot;[%WINDIR]\System32\icacls.exe&quot; &quot;[%ProgramData]\PuppetLabs\puppet\*&quot; /reset /T /C"
+        Execute="immediate"/>
+    <CustomAction
+        Id="ResetPuppetPermissions"
+        BinaryKey="WixCA"
+        DllEntry="<%= @platform.architecture == "x64" ? "WixQuietExec64" : "WixQuietExec" %>"
+        Execute="deferred"
+        Return="check"
+        Impersonate="no"/>
+
+    <CustomAction
```

Comment:
This `CustomAction` and the one below are both MCO related and can be removed here